src/main/java/com/gregtechceu/gtceu/api/machine/feature/ICopyable.java

```diff
@@ -0,0 +1,22 @@
+package com.gregtechceu.gtceu.api.machine.feature;
+
+import net.minecraft.nbt.CompoundTag;
+import net.minecraft.network.chat.Component;
+import net.minecraft.world.item.Item;
+
+import org.jetbrains.annotations.Nullable;
+
+import java.util.Map;
+
+public interface ICopyable {
+
+    void copyConfig(CompoundTag nbt);
+
+    void pasteConfig(CompoundTag nbt);
+
+    Map<Item, Integer> getItemsRequiredForPaste(CompoundTag nbt);
```

Comment:
this should allow requiring NBT, e.g.
```suggestion
    Set<ItemStack> getItemsRequiredForPaste(CompoundTag nbt);
```

Reply:
also, you should use the fastutil maps when you have primitive keys or values, as they're optimized for the specific primitive type(s).